projects/online/online/utils/dataloading.py

```diff
@@ -0,0 +1,257 @@
+
+    Expects frame names to follow a standard nomenclature
+    where the name of the frame file ends {prefix}_{timestamp}-{length}.gwf
+
+    Args:
+        fname: The name of the frame file
+    Returns:
+        The prefix of the frame file name
+        The initial GPS timestamp of the frame file
+        The length of the frame file in seconds
+    """
+
+    if isinstance(fname, Path):
+        fname = fname.name
+
+    match = fname_re.search(fname)
+    if match is None:
+        raise ValueError(f"Could not parse frame filename {fname}")
+
+    prefix, start, duration, *_ = match.groups()
+    return prefix, int(start), int(duration)
+
+
+def _is_gwf(match):
+    return match is not None and match.group("suffix") == "gwf"
+
+
+def get_prefix(datadir: Path):
+    if not datadir.exists():
+        raise FileNotFoundError(f"No data directory '{datadir}'")
+
+    fnames = map(str, datadir.iterdir())
+    matches = map(fname_re.search, fnames)
+    matches = list(filter(_is_gwf, matches))
+
+    if len(matches) == 0:
+        raise ValueError(f"No valid .gwf files in data directory '{datadir}'")
+
+    t0 = max([int(i.group("start")) for i in matches])
+    prefixes = set([i.group("prefix") for i in matches])
+    if len(prefixes) > 1:
+        raise ValueError(
+            "Too many prefixes {} in data directory '{}'".format(
+                list(prefixes), datadir
+            )
+        )
+
+    durations = set([i.group("duration") for i in matches])
+    if len(durations) > 1:
+        raise ValueError(
+            "Too many lengths {} in data directory '{}'".format(
+                list(durations), datadir
+            )
+        )
+    return list(prefixes)[0], int(list(durations)[0]), t0
+
+
+def reset_t0(datadir, last_t0):
+    tick = time.time()
+    while True:
+        matches = [fname_re.search(i.name) for i in datadir.iterdir()]
+        t0s = np.array([int(i.group("start")) for i in matches if _is_gwf(i)])
+        # t0s = t0s[t0s > last_t0]
+        if t0s.size > 0:
+            t0 = max(t0s)
+            logging.info(f"Resetting timestamp to {t0}")
+            return t0
+
+        time.sleep(1)
+        elapsed = (time.time() - tick) // 1
+        if not elapsed % 10:
+            logging.info(
+                "No new frames available since timestamp {}, "
+                "elapsed time {}s".format(last_t0, elapsed)
+            )
+
+
+def data_iterator(
+    datadir: Path,
+    channel: str,
+    ifos: List[str],
+    sample_rate: float,
+    ifo_suffix: str = None,
+    timeout: Optional[float] = None,
+) -> torch.Tensor:
+    if ifo_suffix is not None:
+        ifo_dir = "_".join([ifos[0], ifo_suffix])
+    else:
+        ifo_dir = ifos[0]
+    prefix, length, t0 = get_prefix(datadir / ifo_dir)
+    middle = "_".join(prefix.split("_")[1:])
+
+    # give ourselves a little buffer so we don't
+    # try to grab a frame that's been filtered out
+    # t0 += length * 2
+    frame_buffer = np.zeros((2, 0))
```

Comment:
We should make this `np.zeros((len(ifos), 0))` to be explicit about where the 2 is coming from

Reply:
@wbenoit26 I agree, if this still makes sense

Reply:
I think so, yes